ajay_workspace/librarymanagement/librarymanagement.py

```diff
@@ -0,0 +1,154 @@
+def generate_book_id():
+    file = open("t1.txt", "r")
+    data = file.readlines()
+    if data:
+        t = data[-1].split(",")
+        new_id = int(t[0])
+        return new_id + 1
+    return 101
+
+
+def add_book(book_id1, book_name, author_name, booked=False):
+    book_id = book_id1
+    book_name = book_name
+    author_name = author_name
+    booked = booked
+    file = open("t1.txt", "a")
+    file.write(f"{book_id},{book_name},{author_name},{booked}\n")
+    print("-" * 20)
+    print("Added successfully...")
+    print("-" * 20)
+    file.close()
+
+
+def borrow_book(book_id):
+    file = open("t1.txt", "r")
```

Comment:
`with open` is a best practice because it ensures that the file is properly closed ,  so you need to do something like this 

read about like how to do file handling with `with open `

Ex :- 
with open('example.txt', 'a') as file:
    #code ...................